src/types.rs

```diff
@@ -382,14 +362,18 @@ impl EventData {
         P: Serialize,
         S: AsRef<str>,
     {
-        let payload = serde_json::to_vec(&payload)?;
-        let payload = Bytes::from(payload);
-        let payload = Payload::Json(payload);
+        let payload = Bytes::from(serde_json::to_vec(&payload)?);
+        let metadata = vec![
+            ("type".to_owned(), event_type.as_ref().to_owned()),
+            ("content-type".to_owned(), "application/json".to_owned()),
+        ]
+        .into_iter()
+        .collect();
```

Comment:
Too much allocation for what we are trying to do. Why not just using a hash map from the get-go?